### ISSUE TITLE
External library causes cacheData undefined when tree shake is enable and target mode is node

### ISSUE THREAD

**Issue by buronnie** [parcel-bundler/parcel#1686]
# 🐛 bug report

When the target is node and tree shake is enabled, if we import an external library, it will cause an error in the packaging phase.

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  "scripts": {
    "build": "parcel build index.js --no-minify --target node",
    "treeshake": "parcel build --experimental-scope-hoisting index.js --no-minify --target node"
  },
  "devDependencies": {
    "parcel-bundler": "^1.9.4"
  },
  "dependencies": {
    "moment": "^2.22.2"
  }
}

```

## 🤔 Expected Behavior

Bundle things up without error.

## 😯 Current Behavior

It throws an error in the packaging phase. 
```
🚨  Cannot read property 'cacheData' of undefined
    at JSConcatPackager.markUsed (/Users/kai.guo/git/kai/parcel-node-target-tree-shake-react/node_modules/parcel-bundler/src/packagers/JSConcatPackager.js:96:18)
    at JSConcatPackager.markUsedExports (/Users/kai.guo/git/kai/parcel-node-target-tree-shake-react/node_modules/parcel-bundler/src/packagers/JSConcatPackager.js:91:12)
    at JSConcatPackager.start (/Users/kai.guo/git/kai/parcel-node-target-tree-shake-react/node_modules/parcel-bundler/src/packagers/JSConcatPackager.js:58:12)
    at Bundle._package (/Users/kai.guo/git/kai/parcel-node-target-tree-shake-react/node_modules/parcel-bundler/src/Bundle.js:190:20)
```

## 💁 Possible Solution

When the target is node, parcel does not bundle external library. So `asset.cacheData.imports` does not include the external library. We need to check if `dep` is `undefined` before we want to call `markUsed`.

## 💻 Code Sample

The issue can be reproduced in this repo:
https://github.com/buronnie/parcel-node-target-tree-shake

Run `npm run build`.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.9.4
| Node             | 8.5.0
| npm/Yarn         | 5.3.0
| Operating System | Mac OS Sierra

**Comment by fathyb** [comment]
Thanks @buronnie for the repro! I submitted a fix in #1730, it's not super clean so it may take a bit to get merged. Please let us know if you still encounter issues with `--target=node` and the PR.

**Comment by buronnie** [comment]
Thanks @fathyb ! The issue has been fixed in your PR.
